debezium-core/src/main/java/io/debezium/util/ConnectionValidationUtil.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.util;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConnectionValidationUtil {
```

Comment:
@rajdangwal Do we need a new class for it? IMHO this would fit well into `Threads` class.
In that case you could also refactor `io.debezium.jdbc.JdbcConnection.doClose()`